loopy/target/c/__init__.py

```diff
@@ -859,7 +859,9 @@ def get_function_declaration(self, codegen_state: CodeGenerationState,
             # subkernel launches occur only as part of entrypoint kernels for now
             from loopy.schedule.tools import get_subkernel_arg_info
             skai = get_subkernel_arg_info(kernel, subkernel_name)
-            passed_names = skai.passed_names
+            passed_names = (skai.passed_names
+                            if self.target.is_executable
```

Comment:
I feel that the real issue is a bit different from what's being handled here:

For GPU-ish targets, a kernel is a collection of "subkernel" enqueues. The interface between the "entrypoint/invoker" and the "subkernels" is internal. Meanwhile, for the C target as used by Firedrake, this distinction is not fully realized at the moment: Users will typically directly call the subkernels, and the "entrypoint/invoker" level just doesn't exist. That makes sense, because there's (always?) only one subkernel, and the invoker, if it existed, would just pass through arguments to the subkernel and call it a day. The flag you're introducing here effectively codifies the ability to skip the "entrypoint/invoker" level.

Could you also add some semblance of this discussion to the docs of `single_subkernel_is_entrypoint`?

Reply:
Added some docs on these lines in 9bfef04.